akka-contrib/src/test/scala/akka/contrib/circuitbreaker/CircuitBreakerProxySpec.scala

```diff
@@ -0,0 +1,536 @@
+      sender.send(circuitBreaker, "test message3")
+
+      receiver.expectMsg("test message1")
+      receiver.expectMsg("test message2")
+      receiver.expectMsg("test message3")
+    }
+
+    "send responses to the right sender" in {
+      val sender1 = TestProbe()
+      val sender2 = TestProbe()
+      val receiver = TestProbe()
+      val circuitBreaker = system.actorOf(baseCircuitBreakerPropsBuilder.props(target = receiver.ref))
+
+      sender1.send(circuitBreaker, "test message1")
+      sender2.send(circuitBreaker, "test message2")
+
+      receiver.expectMsg("test message1")
+      receiver.reply("response1")
+
+      receiver.expectMsg("test message2")
+      receiver.reply("response2")
+
+      sender1.expectMsg("response1")
+      sender2.expectMsg("response2")
+    }
+
+    "return failed responses too" in {
+      val sender = TestProbe()
+      val receiver = TestProbe()
+      val circuitBreaker = system.actorOf(baseCircuitBreakerPropsBuilder.props(target = receiver.ref))
+
+      sender.send(circuitBreaker, "request")
+
+      receiver.expectMsg("request")
+      receiver.reply("FAILURE")
+
+      sender.expectMsg("FAILURE")
+    }
+
+    "enter open state after reaching the threshold of failed responses" in new CircuitBreakerScenario {
+      val circuitBreaker = defaultCircuitBreaker
+
+      (1 to baseCircuitBreakerPropsBuilder.maxFailures) foreach { index ⇒
+        receiverRespondsWithFailureToRequest(s"request$index")
+      }
+
+      circuitBreakerReceivesSelfNotificationMessage()
+
+      sender.send(circuitBreaker, "request in open state")
+      receiver.expectNoMsg
+    }
+
+    "respond with a CircuitOpenFailure message when in open state " in new CircuitBreakerScenario {
+      val circuitBreaker = defaultCircuitBreaker
+
+      (1 to baseCircuitBreakerPropsBuilder.maxFailures) foreach { index ⇒
+        receiverRespondsWithFailureToRequest(s"request$index")
+      }
+
+      circuitBreakerReceivesSelfNotificationMessage()
+
+      sender.send(circuitBreaker, "request in open state")
+      sender.expectMsg(CircuitOpenFailure("request in open state"))
+    }
+
+    "respond with the converted CircuitOpenFailure if a converter is provided" in new CircuitBreakerScenario {
+      val circuitBreaker = system.actorOf(
+        baseCircuitBreakerPropsBuilder
+          .copy(openCircuitFailureConverter = { failureMsg ⇒ s"NOT SENT: ${failureMsg.failedMsg}" })
+          .props(receiver.ref))
+
+      (1 to baseCircuitBreakerPropsBuilder.maxFailures) foreach { index ⇒
+        receiverRespondsWithFailureToRequest(s"request$index")
+      }
+
+      circuitBreakerReceivesSelfNotificationMessage()
+
+      sender.send(circuitBreaker, "request in open state")
+      sender.expectMsg("NOT SENT: request in open state")
+    }
+
+    "enter open state after reaching the threshold of timed-out responses" in new CircuitBreakerScenario {
+      Given("A circuit breaker actor proxying a test probe")
```

Comment:
Not that I really mind but I haven't seen the given-when-then style anywhere else in the akka code base. Shouldn't it at least be used consistently in all test cases in this test if used?